src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2771,13 +2771,26 @@ module Crystal
     end
 
     def visit(node : Rescue)
+      is_exception_type_callback = Proc(Crystal::Type, Bool).new do |type|
+        type.implements?(@program.exception) || type.module?
+      end
```

Comment:
**question:** Is there a particular reason for using a proc here instead of a private helper method? The proc creates a closure, so it needs an extra heap allocation.